mlflow/projects/_project_spec.py

```diff
@@ -1,25 +1,44 @@
 """Internal utilities for parsing MLproject YAML files."""
 
 import os
+import yaml
 
 from six.moves import shlex_quote
 
 from mlflow import data
+from mlflow.utils.exception import ExecutionException
+
+
+def load_project(directory):
+    mlproject_path = os.path.join(directory, "MLproject")
+    # TODO: Validate structure of YAML loaded from the file
+    if os.path.exists(mlproject_path):
+        with open(mlproject_path) as mlproject_file:
+            yaml_obj = yaml.safe_load(mlproject_file.read())
+    else:
+        yaml_obj = {}
+    entry_points = {}
+    for name, entry_point_yaml in yaml_obj.get("entry_points", {}).items():
+        parameters = entry_point_yaml.get("parameters", {})
+        command = entry_point_yaml.get("command")
+        entry_points[name] = EntryPoint(name, parameters, command)
+    conda_path = yaml_obj.get("conda_env")
+    if conda_path:
```

Comment:
What happens if conda_path is specified, but the file does not exist? What kind of error do we get?

Reply:
Good question, we ultimately get [an exception](https://github.com/mlflow/mlflow/pull/218/files#diff-16104674981d54b31e320402a143eb19R201) when `load_conda_env` is called. We could validate here, although we might still want to validate when loading the environment e.g. in case conda.yaml is deleted (or we could allow for an uglier error message in that case, since it seems like an edge case). `conda_path` itself is also used directly when running `conda env create --file <conda_path>` to create the environment.

Reply:
I'll add a check verifying that `conda_path` points to an existing file here (during project-loading) for now.